app/models/user.rb

```diff
@@ -15,12 +17,44 @@ class User < ApplicationRecord
     format: {with: VALID_EMAIL_REGEX},
     uniqueness: {case_sensitive: false}
   validates :password, presence: true,
-    length: {minimum: Settings.model_user.password.minimum}
+    length: {minimum: Settings.model_user.password.minimum},
+    on: :create
```

Comment:
đổi từ allow_blank sang on: :create để nó chi validate khi create

Reply:
chưa đúng lắm nhé, c tìm hiểu thêm đi nhé. bản thân `has_secure_password` đã validate thằng password rồi mà

Reply:
ý t là cái này nó sẽ chỉ yêu cầu validate khi create thôi, còn update nó không yêu cầu

Reply:
lúc đăng ký a thử nhập mật khẩu không hợp lệ xem kết quả như thế nào ạ?

Reply:
nó có báo lỗi mật khẩu không hợp lệ mà

Reply:
trường hợp không nhập mật khẩu ấy a

Reply:
The form contains 3 errors.
Password can't be blank
Password is too short (minimum is 6 characters)
Password can't be blank
nó báo như thế này

Reply:
đúng rồi a. trước tiên là a xem nguyên nhân tại sao nó báo 2 lần ạ. rồi sau đó giải quyết như thế nào?

Reply:
1 cái do viết validates cho password nó báo ,còn 1 cái là do has_secure_password nó báo,
ban đầu a dùng allow_blank hoặc allow_nil thay cho on: :create thì nó chỉ báo 1 lần , vì nó sẽ bỏ qua validate mà has_secure_password yêu cầu ,ko biết dùng cách này dc ko hay phải dùng cách khác nữa

Reply:
1 cái do viết validates cho password nó báo ,còn 1 cái là do has_secure_password nó báo,
ban đầu a dùng allow_blank hoặc allow_nil thay cho on: :create thì nó chỉ báo 1 lần , vì nó sẽ bỏ qua validate mà has_secure_password yêu cầu ,ko biết dùng cách này dc ko hay phải dùng cách khác nữa

Reply:
dùng allow_nil nhé a.